File: ros2_ws/src/my_cpp_pkg/src/robot_news_station.cpp
#include <chrono>
#include <string_view>

#include "example_interfaces/msg/string.hpp"
#include "rclcpp/rclcpp.hpp"

namespace {
std::string_view sv_name{"robot_news_station"};
}

class RobotNewsStationNode : public rclcpp::Node {
 public:
  RobotNewsStationNode() : Node(sv_name.data()) {
    publisher_ = this->create_publisher<example_interfaces::msg::String>(
        "robot_news", 10);
    timer_ = this->create_wall_timer(
        std::chrono::seconds(1),
        std::bind(&RobotNewsStationNode::publishNews, this));
    RCLCPP_INFO(this->get_logger(), "%s has started...", sv_name.data());
  }

 private:
  void publishNews(void) {
    auto msg = example_interfaces::msg::String();
    msg.data = std::string("Hi, this is ") + robot_name + std::string(" from the robot News Station");
    publisher_->publish(msg);
  }
  const std::string robot_name{"R2D2"};
  rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv) {
  rclcpp::init(argc, argv);
  auto node = std::make_shared<RobotNewsStationNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class RobotNewsStationNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.robot_name = 'C3PO'
        self.publisher_ = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(1, self.publish_news)
        self.get_logger().info("robot_news_station has started...")

    def publish_news(self):
        msg = String()
        msg.data = f'msg.data being published by {self.robot_name}'
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()